lib/runtime/router.ts

```diff
@@ -251,6 +251,28 @@ export default class Router extends DenaliObject implements RouterDSL {
   }
 
   /**
+   * Returns the URL for a given action. You can supply a params object which
+   * will be used to fill in the dynamic segements of the action's route (if
+   * any).
+   */
+  urlFor(action: string | Action, data: Object): string | null {
```

Comment:
As a heads up, using the `Object` type isn't super useful. It basically implies that the variable has all the methods on the `Object` prototype (i.e. `hasOwnProperty`, etc), _but it cannot accept new properties_. I.e. if you tried to assign `data.foo` in this method, that would be a compiler error.

If you want a POJO (i.e. a JS object that you can dump any old properties onto), the best I've found is to just use `any`. It sucks because you lose _all_ type info (i.e. `2` is a valid `any` type, but clearly isn't what you want). But the more correct alternative is `{ [key: string]: any }`, which is tiresome to type everywhere. So I usually just land on `any`.

The `Object` here works because you just pass it into the `route.reverse` method, which takes a single param with the `{ [key: string]: any }` type, and technically, `new Object()` satisfies that.